File: PL/Scripts/Empleado.js
$(document).ready(function () {
    GetAll();
    EstadoGetAll();
    $('#btnAgregar').click(function () {
        $('#Formulario').modal('show');
    });
});


function GetAll() {
    $.ajax({
        type: 'GET',
        url: 'http://localhost:16942/api/Empleado/GetAll',
        success: function (result) {
            $('#TableEmpleado tbody').empty();
            $.each(result, function (i, empleado) {
                var filas =
                    '<tr>' +
                    '<td class="text-center">' +
                    '<a class="btn btn-warning glyphicon glyphicon-edit" href="#" onclick="GetById(' + empleado.IdEmpleado + ')">' + '</a>' +
                    '</td>' +
                    "<td class='text-center'>" + empleado.IdEmpleado + "</td>" +
                    "<td class='text-center'>" + empleado.NumeroNomina + "</td>" +
                    "<td class='text-center'>" + empleado.Nombre + "</td>" +
                    "<td class='text-center'>" + empleado.ApellidoPaterno + "</td>" +
                    "<td class='text-center'>" + empleado.ApellidoMaterno + "</td>" +
                    "<td class='text-center'>" + empleado.Estado.IdEstado + "</td>"
                    + '<td class="text-center"> <button class="btn btn-danger" id="btnEliminar"><span class="glyphicon glyphicon-trash" style="color:#FFFFFF"></span></button></td>'
                    + '</tr>';
                $("#TableEmpleado tbody").append(filas);
            });
        },
        error: function (result) {
            alert('Error en la consulta. ' + result.responseJSON.result);
        }
    });
};

function EstadoGetAll() {
    $("#ddlEstados").empty();
    $.ajax({
        type: 'GET',
        url: 'http://localhost:16942/api/Estado/GetAll',
        success: function (result) {
            $("#ddlEstados").append('<option value="' + 0 + '">' + 'Seleccione una opcion' + '</option>');
            $.each(result, function (i, estado) {
                $("#ddlEstados").append('<option value="'
                    + estado.IdEstado + '">'
                    + estado.Nombre + '</option>');
            });
        }
    });
};

//function Guardar() {

//    var empleado = $('#txtIdEmpleado').val();
//    if (txtIdEmpleado == "") {
//        Add(empleado);
//    }
//    else {
//        Update(empleado);
//    }
//}


//function Add(empleado) {

//    var empleado = {
//        IdEmpleado: 0,
//        NumeroNomina: $('txtNumeroNomina').val(),
//        Nombre: $('txtNombre').val(),
//        ApellidoPaterno: $('txtApellidoPaterno').val(),
//        ApellidoMaterno: $('txtApellidoMaterno').val(),
//        Estado: {
//            IdEstado: $('#ddlEstados').val(0)
//        }
//    } 


function Add(empleado) {

    $.support.cors = true;
    $.ajax({
        type: 'POST',
/*        crossDomain: true,*/
        url: 'http://localhost:16942/api/Empleado/Add',
        dataType: 'json',
        data: JSON.stringify(empleado),
        contentType: 'application/json; charset=utf-8',
 /*       data: empleado,*/
        success: function (result) {
            $('#myModal').modal();
            $('#ModalUpdate').modal('hide');
            GetAll();
        },
        //error: function (result) {
        //    alert('Error' + result.responseJSON.ErrorMessage);
        //}
    });
};


function Update(empleado) {
    $.ajax({
        type: 'POST',
        url: 'http://localhost:16942/api/Empleado/Update',
        dataType: 'json',
        data;
    })
}

function Modal() {
    var mostrar = $('#ModalUpdate').modal('show');
    IniciarEmpleado();

}

function Actualizar() {
    var empleado = {
        IdEmpleado: $('#txtIdEmpleado').val(),
        NumeroNomina: $('#txtNumeroNomina').val(),
        Nombre: $('#txtNombre').val(),
        ApellidoPaterno: $('#txtApellidoPaterno').val(),
        ApellidoMaterno: $('#txtApellidoMaterno').val(),
        Estado: {
            IdEstado: $('#ddlEstados').val()
        }
    }

    if (empleado.IdEmpleado == '') {
        Add(empleado);

    }
    else {
        Update(empleado);
    }
}

function IniciarEmpleado() {

    var empleado = {
        IdEmpleado: $('#txtIdEmpleado').val(''),
        NumeroNomina: $('#txtNumeroNomina').val(''),
        Nombre: $('#txtNombre').val(''),
        ApellidoPaterno: $('#txtApellidoPaterno').val(''),
        ApellidoMaterno: $('#txtApellidoMaterno').val(''),
        Estado: {
            IdEstado: $('#ddlEstados').val(0)
        }
    }
}

File: BL/Empleado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BL
{
    public class Empleado
    {
        public static ML.Result GetAll()
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.FMolinaLeenkenGroupEntities context = new DL.FMolinaLeenkenGroupEntities())
                {
                    var query = context.EmpleadoGetAll().ToList();
                    result.Objects = new List<object>();

                    if (query != null)
                    {
                        foreach (var obj in query)
                        {
                            ML.Empleado empleado = new ML.Empleado();

                            empleado.IdEmpleado = obj.IdEmpleado;
                            empleado.NumeroNomina = obj.NumeroNomina;
                            empleado.Nombre = obj.Nombre;
                            empleado.ApellidoPaterno = obj.ApellidoPaterno;
                            empleado.ApellidoMaterno = obj.ApellidoMaterno;

                            empleado.Estado = new ML.Estado();
                            empleado.Estado.IdEstado = obj.IdEstado;

                            result.Objects.Add(empleado);
                        }
                        result.Correct = true;
                    }
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }

        public static ML.Result Add(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.FMolinaLeenkenGroupEntities context = new DL.FMolinaLeenkenGroupEntities())
                {
                    var query = context.EmpleadoAdd(empleado.NumeroNomina, empleado.Nombre, empleado.ApellidoMaterno, empleado.ApellidoMaterno, empleado.Estado.IdEstado);
                    if (query > 0)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "Error al agregar";
                    }
                    result.Correct = true;
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }

        public static ML.Result Update(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.FMolinaLeenkenGroupEntities context = new DL.FMolinaLeenkenGroupEntities())
                {
                    var query = context.EmpleadoUpdate(empleado.IdEmpleado, empleado.NumeroNomina, empleado.Nombre, empleado.ApellidoPaterno, empleado.ApellidoMaterno, empleado.IdEstado);

                    if (query > 0)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                        result.ErrorMessage = "Error al actualizar";
                    }
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }

        public static ML.Result GetById(int IdEmpleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.FMolinaLeenkenGroupEntities context = new DL.FMolinaLeenkenGroupEntities())
                {
                    var query = context.EmpleadoGetById(IdEmpleado).FirstOrDefault();

                    if (query != null)
                    {
                        ML.Empleado empleado = new ML.Empleado();

                        empleado.IdEmpleado = query.IdEmpleado;
                        empleado.NumeroNomina = query.NumeroNomina;
                        empleado.Nombre = query.Nombre;
                        empleado.ApellidoPaterno = query.ApellidoPaterno;
                        empleado.ApellidoMaterno = query.ApellidoMaterno;
                        empleado.IdEstado = query.IdEstado.Value;

                        result.Correct = true;

                    }
                    else
                    {
                        result.Correct = false;
                    }
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }

        public static ML.Result Delete(ML.Empleado empleado)
        {
            ML.Result result = new ML.Result();
            try
            {
                using (DL.FMolinaLeenkenGroupEntities context = new DL.FMolinaLeenkenGroupEntities())
                {
                    var query = context.EmpleadoDelete(empleado.IdEmpleado);
                    if (query > 0)
                    {
                        result.Correct = true;
                    }
                    else
                    {
                        result.Correct = false;
                    }
                }
            }
            catch (Exception ex)
            {
                result.Correct = false;
                result.ErrorMessage = ex.Message;
            }
            return result;
        }
    }
}
